### PR TITLE
Update vectorgeneral.rst

### PR DESCRIPTION
**Pull request by rosaguilar**
Documenting Repair Shapefile Algorithm

Goal:

Ticket(s): #4553

- [ ] Backport to LTR documentation is required

### CODE REVIEW

File: docs/user_manual/processing_algs/qgis/vectorgeneral.rst [hunk 1]
```diff
@@ -1535,7 +1535,47 @@ Outputs
...
+   :stub-columns: 0
+
+   * - Label
+     - Name
+     - Type
+     - Description
+   * - 
+     - ``INPUT``
+     - [same as input]
+     - The input vector layer with the SHX files repaired
```
Comment (havatv) [inline docs/user_manual/processing_algs/qgis/vectorgeneral.rst]:
```suggestion
     - The input vector layer with the SHX file repaired
```

File: docs/user_manual/processing_algs/qgis/vectorgeneral.rst [hunk 2]
```diff
@@ -1535,7 +1535,47 @@ Outputs
      - [same as input]
      - Output (sorted) vector layer
 
+.. _qgisrepairshapefile:
 
+Repair Shapefile
+----------------
+Repairs a broken Shapefile by recreating missing or broken SHX files.
```
Comment (havatv) [inline docs/user_manual/processing_algs/qgis/vectorgeneral.rst]:
```suggestion
Repairs a broken ESRI Shapefile dataset by (re)creating the SHX file.
```

File: docs/user_manual/processing_algs/qgis/vectorgeneral.rst [hunk 3]
```diff
@@ -1535,7 +1535,47 @@ Outputs
...
+.. list-table::
+   :header-rows: 1
+   :widths: 20 20 20 40
+   :stub-columns: 0
+
+   * - Label
+     - Name
+     - Type
+     - Description
+   * -
```
Comment (havatv) [inline docs/user_manual/processing_algs/qgis/vectorgeneral.rst]:
```suggestion
   * - **Repaired layer**
```

File: docs/user_manual/processing_algs/qgis/vectorgeneral.rst [hunk 4]
```diff
@@ -1535,7 +1535,47 @@ Outputs
...
+   :header-rows: 1
+   :widths: 20 20 20 40
+   :stub-columns: 0
+
+   * - Label
+     - Name
+     - Type
+     - Description
+   * - 
+     - ``INPUT``
```
Comment (havatv) [inline docs/user_manual/processing_algs/qgis/vectorgeneral.rst]:
```suggestion
     - ``OUTPUT``
```

File: docs/user_manual/processing_algs/qgis/vectorgeneral.rst [hunk 5]
```diff
@@ -1535,7 +1535,47 @@ Outputs
...
+   :widths: 20 20 20 40
+   :stub-columns: 0
+
+   * - Label
+     - Name
+     - Type
+     - Description
+   * - **Input Shapefile**
+     - ``INPUT``
+     - [vector: shapefile]
```
Comment (havatv) [inline docs/user_manual/processing_algs/qgis/vectorgeneral.rst]:
```suggestion
     - [file]
```

File: docs/user_manual/processing_algs/qgis/vectorgeneral.rst [hunk 6]
```diff
@@ -1535,7 +1535,47 @@ Outputs
...
+     - Full path to the vector layer with missing or broken SHX files
```
Comment (havatv) [inline docs/user_manual/processing_algs/qgis/vectorgeneral.rst]:
```suggestion
     - Full path to the ESRI Shapefile dataset with a missing or broken SHX file
```

File: docs/user_manual/processing_algs/qgis/vectorgeneral.rst [hunk 7]
```diff
@@ -1535,7 +1535,47 @@ Outputs
...
+   :widths: 20 20 20 40
+   :stub-columns: 0
+
+   * - Label
+     - Name
+     - Type
+     - Description
+   * - 
+     - ``INPUT``
+     - [same as input]
```
Comment (havatv) [inline docs/user_manual/processing_algs/qgis/vectorgeneral.rst]:
```suggestion
     - [vector any]
```

**Comment by rosaguilar** [comment]
Thanks for comments and suggestions.

**Comment by havatv** [comment]
> Thanks @havatv for comments and suggestions.

You are welcome! 

Regarding committing suggestions, we can avoid a lot of messages if you

1. Go to the *Files changed* "tab"
2. Go through all the suggestions and press the "Add suggestion to batch" button (if you accept the suggestion)
3. When you have added the suggestions you want to accept, you can commit them by pressing the "Commit suggestions" button

This will collect the suggestions in one commit, and only one message will be sent. https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request

**Comment by havatv** [comment]
Thanks, again, @rosaguilar!

**Comment by rosaguilar** [comment]
> > Thanks @havatv for comments and suggestions.
> 
> You are welcome!
> 
> Regarding committing suggestions, we can avoid a lot of messages if you
> 
> 1. Go to the _Files changed_ "tab"
> 2. Go through all the suggestions and press the "Add suggestion to batch" button (if you accept the suggestion)
> 3. When you have added the suggestions you want to accept, you can commit them by pressing the "Commit suggestions" button
> 
> This will collect the suggestions in one commit, and only one message will be sent. https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request

Thanks. Good to know.

**Pull request merged**